### PR TITLE
Rewrite xrfi_run more cleanly and concisely

### PR DESCRIPTION
**Pull request by jsdillon**
By changing what's looped over and what's explicitly written out and using a dictionary for repeated kwargs, I've vastly simplified `xrfi_run` without changing its inputs or outputs. Hopefully, this will make things easier to debug.

### CODE REVIEW

**Review by aewallwi — CHANGES_REQUESTED** [review]
Make sure to perform a comparitive test to ensure that the flags and metrics from the rewritten version of this function match the flags from the old version of this function.

**Comment by jsdillon** [comment]
image

I've run XRFI on both this branch and the master branch for a subset of 2459122 with the same settings used for that day. I get the same results (except in the history field, of course), so I'm feeling pretty confident I haven't introduced any bugs.

File: hera_qm/xrfi.py [hunk 1]
```diff
@@ -1998,254 +2003,154 @@ def xrfi_run(ocalfits_files=None, acalfits_files=None, model_files=None,
...
-        if len(flags) > 1:
-            for flg in flags[1:]:
-                vdict['uvf_init'] |= flg
-        vdict['uvf_init'].label = 'ORd flags, round 1.'
 
+    def _run_all_filters(median_round, 
+                         omnical_filter, omnical_chi2_filter, omnical_zscore_filter, 
+                         abscal_filter, abscal_chi2_filter, abscal_zscore_filter,
+                         omnivis_filter, cross_filter, auto_filter):
+        '''This function runs all possible filters, updating vdict as appropriate'''
```
[the remark this answers is not in the document]
Comment (aewallwi) [reply hera_qm/xrfi.py:2024]: Doesn't need to be a full fledged one, but just a sentence describing the nature of all the args (they are switches to run the different types of filter/flag step).

File: hera_qm/xrfi.py [hunk 2]
```diff
@@ -1998,254 +2003,154 @@ def xrfi_run(ocalfits_files=None, acalfits_files=None, model_files=None,
...
+        
+        # Start with empty list of flags and metrics for later combination
+        metrics = []
+        flags = []
+        
+        # Modify these labels and parameters based on whether we're doing mean- or median-based statistics
+        label = {True: 'median', False: 'mean'}[median_round]
```
Comment (aewallwi) [inline hera_qm/xrfi.py:2031]: note the True/False nature of median round in the docstring or use a stringe for median_round

**Review by aewallwi — CHANGES_REQUESTED** [review]
Generally looking pretty good. I think this yielded substantial readability and maintainability improvements. Just a few minor adjustments still. A docstring clarifications in _run_all_filters would be nice. Also, I think it'd be good to check that you get the exact same results from the old unittests in `test_xrfi_run` to the new version -- not that the old one is any more correct but just to make sure we are making controlled changes (preserving the exact old functionality).

File: hera_qm/xrfi.py (hunk 2 above)
Comment (jsdillon) [reply hera_qm/xrfi.py:2031]: Done.

**Review by aewallwi — APPROVED** [review]
Looks good. I approve.

**Pull request merged**